MySampleApp/app/src/main/java/com/mysampleapp/util/SermonHandler.java

```
@@ -0,0 +1,58 @@
+package com.mysampleapp.util;
+
+
+import android.util.Log;
+
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.mobileconnectors.dynamodbv2.dynamodbmapper.DynamoDBScanExpression;
+
+import java.util.Date;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import com.amazonaws.mobile.AWSMobileClient;
+import com.amazonaws.models.nosql.SermonsDO;
+
+public class SermonHandler extends DatabaseHandler{
+
+    public void save(Sermon sermon) throws AmazonClientException{
+        AmazonClientException lastException = null;
+
+        final SermonsDO sermonItem = new SermonsDO();
+        sermonItem.setUserId(AWSMobileClient.defaultMobileClient().getIdentityManager().getCachedUserID());
+        sermonItem.setCreationDate((double) sermon.getDate().getTime());
+        sermonItem.setTitle(sermon.getTitle());
+        sermonItem.setSermon(sermon.getSermonURL());
+        sermonItem.setVerse(sermon.getVerse());
+        sermonItem.setSeries(sermon.getSeries());
+
+        try {
+            super.save(sermonItem);
+        } catch (final AmazonClientException ex) {
+            lastException = ex;
```

Instead of saving the Exception in `lastException` and throw later, can we throw the error right away?